src/Models/HydrostaticFreeSurfaceModels/SplitExplicitFreeSurfaces/split_explicit_free_surface.jl

```diff
@@ -0,0 +1,318 @@
+                      occurs and ``M_*`` corresponds to the last barotropic time step for which the
+                      `averaging_kernel > 0`. By default, the averaging kernel described by [Shchepetkin2005](@citet)
+                      is used.
+
+- `timestepper`: Time stepping scheme used for the barotropic advancement. Choose one of:
+  * `ForwardBackwardScheme()` (default): `η = f(U)`   then `U = f(η)`,
+  * `AdamsBashforth3Scheme()`: `η = f(U, Uᵐ⁻¹, Uᵐ⁻²)` then `U = f(η, ηᵐ, ηᵐ⁻¹, ηᵐ⁻²)`.
+
+References
+==========
+
+Shchepetkin, A. F., & McWilliams, J. C. (2005). The regional oceanic modeling system (ROMS): a split-explicit, free-surface, topography-following-coordinate oceanic model. Ocean Modelling, 9(4), 347-404.
+"""
+function SplitExplicitFreeSurface(grid = nothing;
+                                  gravitational_acceleration = g_Earth,
+                                  substeps = nothing,
+                                  cfl = nothing,
+                                  fixed_Δt = nothing,
+                                  averaging_kernel = averaging_shape_function,
+                                  timestepper = ForwardBackwardScheme())
+
+    if !isnothing(grid)
+        FT = eltype(grid)
+    else
+        # this is a fallback and only used via the outer constructor,
+        # in case no grid is provided; when afterwards the free surfade
+        # is materialized via materialize_free_surface
+        # FT becomes eltype(grid)
+        FT = Float64
+    end
+
+    gravitational_acceleration = convert(FT, gravitational_acceleration)
+    substepping = split_explicit_substepping(cfl, substeps, fixed_Δt, grid, averaging_kernel, gravitational_acceleration)
+
+    kernel_parameters = :xy 
+
+    return SplitExplicitFreeSurface(nothing,
+                                    nothing,
+                                    nothing,
+                                    gravitational_acceleration,
+                                    kernel_parameters,
+                                    substepping,
+                                    timestepper)
+end
+
+# Simplest case: we have the substeps and the averaging kernel
+function split_explicit_substepping(::Nothing, substeps, fixed_Δt, grid, averaging_kernel, gravitational_acceleration)
+    FT = eltype(gravitational_acceleration)
+    fractional_step_size, averaging_weights = weights_from_substeps(FT, substeps, averaging_kernel)
+    return FixedSubstepNumber(fractional_step_size, averaging_weights)
+end
+
+# The substeps are calculated dynamically when a cfl without a fixed_Δt is provided
+function split_explicit_substepping(cfl, ::Nothing, ::Nothing, grid, averaging_kernel, gravitational_acceleration)  
+    if isnothing(grid)
+        throw(ArgumentError(string("Need to provide the grid to calculate the barotropic substeps from the cfl. ",
+                                    "For example, SplitExplicitFreeSurface(grid, cfl=0.7, ...)")))
```

Comment:
```suggestion
        throw(ArgumentError(string("The grid is a required positional argument to SplitExplicitFreeSurface when cfl is specified.",
                                    "For example, SplitExplicitFreeSurface(grid, cfl=0.7, ...)")))
```

However, I think it is better to do this validation in the constructor. Notice that the error really depends on what the constructor looks like. If somebody tries to change the constructor in the future, they may have to change this code as well. Therefore, it belongs in the constructor where the function signature that is required is defined.